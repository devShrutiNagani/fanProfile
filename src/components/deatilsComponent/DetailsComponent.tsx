import {View, Text} from 'react-native';
import React from 'react';
import createStyles from './Styles';
import {CustomDetailsComponent} from '../../constant/interface/custom/Custom';

const DetailsComponent: React.FC<CustomDetailsComponent> = ({title, data}) => {
  const styles = createStyles();
  return (
    <View style={styles.textView}>
      <Text style={styles.titleText}>{title}</Text>
      <Text style={styles.text}>{data}</Text>
    </View>
  );
};
export default DetailsComponent;
