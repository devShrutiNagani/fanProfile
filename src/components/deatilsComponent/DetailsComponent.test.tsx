import React from 'react';
import {render} from '@testing-library/react-native';
import DetailsComponent from './DetailsComponent';

describe('DetailsComponent', () => {
  it('renders the title and data correctly', () => {
    const {getByText} = render(
      <DetailsComponent title="Test Title" data="Test Data" />,
    );

    expect(getByText('Test Title')).toBeTruthy();
    expect(getByText('Test Data')).toBeTruthy();
  });

  it('renders empty text when no data is provided', () => {
    const {getByText} = render(<DetailsComponent title="Only Title" data="" />);

    expect(getByText('Only Title')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = render(
      <DetailsComponent title="Snapshot Title" data="Snapshot Data" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
