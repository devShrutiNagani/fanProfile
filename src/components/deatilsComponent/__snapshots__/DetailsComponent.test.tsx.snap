// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsComponent matches the snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginBottom": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "#1D1D1D",
        "fontFamily": "Montserrat-SemiBold",
        "fontSize": 24,
        "includeFontPadding": false,
      }
    }
  >
    Snapshot Title
  </Text>
  <Text
    style={
      {
        "color": "#1D1D1D",
        "fontFamily": "Montserrat-Regular",
        "fontSize": 24,
        "includeFontPadding": false,
      }
    }
  >
    Snapshot Data
  </Text>
</View>
`;
